RecoParticleFlow/PFClusterProducer/plugins/Basic2DGenericTopoClusterizer.cc

```diff
@@ -53,15 +53,23 @@ buildTopoCluster(const edm::Handle<reco::PFRecHitCollection>& input,
       std::abs(cell.positionREP().eta()) > 0.34 ) {
       cell_layer *= 100;
     }    
-  const std::pair<double,double>& thresholds =
-      _thresholds.find(cell_layer)->second;
-  if( cell.energy() < thresholds.first || 
-      cell.pt2() < thresholds.second ) {
-    LOGDRESSED("GenericTopoCluster::buildTopoCluster()")
-      << "RecHit " << cell.detId() << " with enegy " 
-      << cell.energy() << " GeV was rejected!." << std::endl;
-    return;
+
+  std::tuple<std::vector<int> ,std::vector<double> , std::vector<double> > thresholds = _thresholds.find(cell_layer)->second;
```

Comment:
I think that a similar approach should be made here as for the energyNorm.
keep the map value types to be pair<double,double>
and extend the meaning of the index to be able to directly map (cell_layer, depth) pair to an int.